test/unit/test_flags.py

```diff
@@ -66,13 +67,13 @@ def test__flags(self):
         # warn_error_options
         self.user_config.warn_error_options = '{"include": "all"}'
         flags.set_from_args(self.args, self.user_config)
-        self.assertEqual(flags.WARN_ERROR_OPTIONS, '{"include": "all"}')
+        self.assertEqual(flags.WARN_ERROR_OPTIONS, WarnErrorOptions(include="all"))
```

Comment:
We've spec'd out the include/exclude spec [as the result of this discussion](https://github.com/dbt-labs/dbt-core/issues/6165#issuecomment-1371289105) to be `{"include": str | list, "exclude": list}`, and implemented [here](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/helper_types.py#L96). I'm open to revisiting it but that would represent a breaking change and feels out of scope for this PR.

Reply:
resolved: making a follow-up issue to use an Enum instead of a string in the `IncludeExclude` dataclass to represent the "all"/"*" options of the `include` attribute.

Reply:
https://github.com/dbt-labs/dbt-core/issues/6786